### DISCUSSION TITLE
compile loop

### DISCUSSION THREAD

**Discussion by Paolo-lab** [iTwin/itwinjs-core#6007] [Q&A] [answered]
Hi everyone, I had a description problem which thanks @aruniverse I solved by switching from version "@bentley/react-scripts": "^5.0.1" to version 5.0.5.
Now I have the problem that the compilation of my application never stops and keeps repeating itself endlessly

I attach the package.json maybe someone can advise me on some package that I should update

ps I also added "USING_NPM=true" to the .env file, but the result is the same.

Node version is v16.17.0

tanks all.

[package.zip](https://github.com/iTwin/itwinjs-core/files/12649469/package.zip)

**Comment by Paolo-lab** [comment]
Hi everyone, good morning, I think I've run out of options, I've updated node and NPM to the versions
node : v18.18.0
NPM : 9.8.1
I deleted the node_modules folder and the package-lock.json file

I installed everything from scratch with "npm i" but the result doesn't change I continue to have an infinite loop on the compilation, so to speak the program runs the same, but suffers from a slowdown.

I also tried to recreate a new application with
npx create-react-app reactapp4 --template @itwin/web-viewer --scripts-version @bentley/react-scripts , which installed version 4 for me.
but even here without adding anything I have the same problem during compilation.. it goes in a continuous loop!!!

I don't know what to do anymore...
 
I hope you can give me some pointers
one thing I haven't tried is taking the application to another computer and attempting to compile it

Thanks for your precious help

**Reply by AnDuong249** [reply]
Hi there,

Would you mind providing a screenshot of the terminal and the console of the app in the developer tools when that bug is happening? Were you running npm run start or build when this bug happen? 

I tried creating a new application with your command "npx create-react-app reactapp4 --template @itwin/web-viewer --scripts-version @bentley/react-scripts" with this environment (node : v18.18.0, NPM : 9.8.1) but I can't seem to reproduce what you're mentioning.

Thanks!

**Comment by Paolo-lab** [comment]
hi, I have the problem when I do npm start, when I do npn run build everything works.

It might depend on webpack, the version that is automatically installed is 5.88.2 which is the latest release available

do you think it is advisable to try installing the package through YARN rather than NPM? I attach the images of the console in the development tools I have nothing to show because it doesn't give me any errors inherent to the problem
It's not clear from the image I'm attaching, but basically it repeats the image I'm sending you endlessly..

thank you for your precious help
<img width="311" alt="loop compile" src="https://github.com/iTwin/itwinjs-core/assets/57446652/f45f4631-f39a-46d6-a9ea-fba919fac5f5">

**Reply by AnDuong249** [reply]
Hi there,

Thank you for the response. Can you provide the create-react-app version that you were using with this command "npx create-react-app --version"? Also, would you mind testing with the latest create-react-app version using the command "npx create-react-app@latest reactapp4 --template @itwin/web-viewer --scripts-version @bentley/react-scripts" to see if there is any differences.

If that doesn't have any differences, in the meantime, you can try running yarn install and yarn start to see if that also shows up as well.

Thanks!

**Comment by Paolo-lab** [comment, marked as answer]
Hi @AnDuong249, sorry for replying only now, I don't know why all this is happening, I took my application and brought it to another computer and everything works fine, it seems that my laptop has some problems in the next few days I would like to check the size of node_modules of the two machines and try to understand from there what problems I have for now thank you for your attention

thanks Paul
